questions/Predict the output

```diff
@@ -0,0 +1,30 @@
+### // interviewer: what will the following code output?
+'const arr = [10, 12, 15, 21];
+for (var i = 0; i < arr.length; i++) {
+  setTimeout(function() {
+    console.log('Index: ' + i + ', element: ' + arr[i]);
+  }, 3000);
+}'
+
+#### Answer
+
+Index: 4, element: undefined(printed 4 times).
+
+In each iteration, control goes to timeout method and wait for 3000ms to be completed.
```

Comment:
To me, this answer is incomplete and wouldn't show an understanding of immutability, hoisting, scopes, and closure to an interviewer. 

In order to merge it, the answer would need a complete rewrite.